pkg/service/test/test.go

```diff
@@ -333,10 +337,14 @@ func (t *tester) Test(path string, testReportPath string, generateTestReport boo
 		}
 	}
 
+
+	// Remove the container on program exit
+	removeContainer := !strings.Contains(appCmd, "docker start")
+
 	if !initialisedValues.AbortStopHooksForcefully {
 		initialisedValues.AbortStopHooksInterrupt <- true
 		// stop listening for the eBPF events
-		initialisedValues.LoadedHooks.Stop(true)
+		initialisedValues.LoadedHooks.Stop(true, removeContainer)
```

Comment:
This is not a preferred approach, We shouldn't signal things from service layer a lot.